jfr-mappers/src/main/java/com/newrelic/jfr/tosummary/BasicGarbageCollectionSummarizer.java

```diff
@@ -0,0 +1,141 @@
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+import jdk.jfr.consumer.RecordedEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** This class aggregates the duration of Garbage Collection JFR events */
+public final class BasicGarbageCollectionSummarizer implements EventToSummary {
+  private static final Logger logger =
+      LoggerFactory.getLogger(BasicGarbageCollectionSummarizer.class);
+
+  public static final String EVENT_NAME = "jdk.GarbageCollection";
+
+  private final SimpleDurationSummarizer minorGcDurationSummarizer;
+  private final SimpleDurationSummarizer majorGcDurationSummarizer;
+  private int minorGcCount = 0;
+  private int majorGcCount = 0;
+  private long startTimeMs;
+  private long minorGcEndTimeMs = 0L;
+  private long majorGcEndTimeMs = 0L;
+
+  private static final Set<String> MINOR_GC_NAMES =
+      Collections.unmodifiableSet(
+          new HashSet<String>() {
+            {
+              add("DefNew");
+              add("G1New");
+              add("ParallelScavenge");
+              add("ParNew");
+              add("PSMarkSweep");
+            }
+          });
+
+  private static final Set<String> MAJOR_GC_NAMES =
+      Collections.unmodifiableSet(
+          new HashSet<String>() {
+            {
+              add("ConcurrentMarkSweep");
+              add("G1Full");
+              add("G1Old");
+              add("ParallelOld");
+              add("SerialOld");
+            }
+          });
+
+  public BasicGarbageCollectionSummarizer() {
+    this(Instant.now().toEpochMilli());
+  }
+
+  public BasicGarbageCollectionSummarizer(long startTimeMs) {
+    this(
+        startTimeMs,
+        new SimpleDurationSummarizer(startTimeMs, DEFAULT_CLOCK, "duration"),
+        new SimpleDurationSummarizer(startTimeMs, DEFAULT_CLOCK, "duration"));
+  }
+
+  public BasicGarbageCollectionSummarizer(
+      long startTimeMs,
+      SimpleDurationSummarizer minorGcDurationSummarizer,
+      SimpleDurationSummarizer majorGcDurationSummarizer) {
+    this.startTimeMs = startTimeMs;
+    this.minorGcDurationSummarizer = minorGcDurationSummarizer;
+    this.majorGcDurationSummarizer = majorGcDurationSummarizer;
+  }
+
+  @Override
+  public String getEventName() {
+    return EVENT_NAME;
+  }
+
+  @Override
+  public void accept(RecordedEvent ev) {
+    final String name = ev.getValue("name");
+    if (name != null) {
+      if (MINOR_GC_NAMES.contains(name)) {
+        minorGcEndTimeMs = ev.getStartTime().toEpochMilli();
+        minorGcDurationSummarizer.accept(ev);
+        minorGcCount++;
+      } else if (MAJOR_GC_NAMES.contains(name)) {
+        majorGcEndTimeMs = ev.getStartTime().toEpochMilli();
+        majorGcDurationSummarizer.accept(ev);
+        majorGcCount++;
+      } else
+        // Ignore events with GC name: GCNameEndSentinel, N/A, Shenandoah, Z or anything unexpected
+        logger.warn("Ignoring unsupported " + EVENT_NAME + " event: " + name);
```

Comment:
@kittylyst You mentioned wanting to log when we skip an event with an unsupported GC name, so I added this. However, there was no prior precedent of doing any logging from the jfr-mappers module which means that the slf4j dependency had to be added. Is this definitely something that we want to do? If not, just let me know and I can roll back the logging changes.

Reply:
I had wanted to keep the transformation library free from logging. However, we are still tightly coupled to the telemetry library, so I'm not sure that matters that much right now. We can revisit later - for now, let's add the dependency